Update axum to 0.6.1 and update other packages

romanstingler opened the pull request:
<!--
Thank you for submitting to the TechEmpower Framework Benchmarks!

If you are submitting a new framework, please make sure that you add the appropriate line in the `.github/workflows/build.yml` file for proper integration testing. Also please make sure that an appropriate `README.md` is added in your framework directory with information about the framework and a link to its homepage and documentation.

For new frameworks, please do not include source code that isn't required for the benchmarks.

Some examples of files that should not be included:

* Functional tests, such as JUnit tests in a `src/test` directory in Java frameworks.
* Startup scripts for launching the framework's application directly without going through TFB.
* Local development configs used on the developer's workstation but not in TFB.

If you are editing an existing test, please update the `README.md` for that test where appropriate.
-->

romanstingler commented:
@jplatte hey I updated axum and the other deps maybe you can have a look at it.
I just started this week with rust.
So it would be beneficial if someone could check it.

jplatte commented on frameworks/Rust/axum/Cargo.toml:
```diff
@@ -29,27 +29,37 @@ name = "axum-pg"
+
 
 [profile.release]
+opt-level = 3
+debug = false
+split-debuginfo = 'off'
+debug-assertions = false
+overflow-checks = false
 lto = true
+panic = 'unwind'
+incremental = true
 codegen-units = 1
+rpath = false
+strip = "symbols"
```
Why all these new settings? Many are set to their default values, and `incremental = true` could actually be based for performance.

romanstingler replied:
ahhh, damn, I forgot to remove them, I just tested different compiler options. I will set them to the previous version with lto and codegen-units.

jplatte commented:
Did you run a `cargo update` inside the directory with `Cargo.toml`? That might upgrade some indirect dependencies (I'm asking because you also bumped many minimum version requirements in `Cargo.toml` for no apparent reason).

jplatte commented on frameworks/Rust/axum/src/database_mongo.rs:
```diff
@@ -14,14 +15,15 @@ use mongodb::Database;
 
 #[async_trait]
-impl<B> FromRequest<B> for DatabaseConnection
+impl<S, B> FromRequest<S, B> for DatabaseConnection
```
All these `FromRequest` implementations should be `FromRequestParts`, because they don't care about the body of the request. Then you can also revert the change in extractor order for the handler functions.

romanstingler replied:
I have pushed the changes FromRequest ->  FromRequestParts.

Yeah, I have checked all included crates for the latest version and updated them.

In my opinion, the benchmarks should represent a performance figure which is representative at the time when the benchmark is executed. 
Meaning, that the latest packages should be used. 
This also gives the possibility to compare benchmark history and in cases of regression maybe find crates that don't have benchmarks in their CI.

There is theoretically a 1.0.0-RC.1 of hyper available, but the latest update was in October, and it dropped 
`hyper::server::conn::AddrIncoming` so there would be more refactoring necessary. 
But as there are updates on the 0.14.x version this should be fine.

 I tested on my local notebook and all tests pass.
 
 I highly appreciate your time and patience with a Rust noob.

jplatte commented on frameworks/Rust/axum/src/main_mongo_raw.rs:
```diff
@@ -35,8 +35,8 @@ async fn db(DatabaseConnection(db): DatabaseConnection) -> impl IntoResponse {
 async fn queries(
-    DatabaseConnection(db): DatabaseConnection,
     Query(params): Query<Params>,
+    DatabaseConnection(db): DatabaseConnection,
```
This change can now be reverted.

jplatte commented on frameworks/Rust/axum/src/main_mongo_raw.rs:
```diff
@@ -56,8 +56,8 @@ async fn queries(
 async fn updates(
-    DatabaseConnection(db): DatabaseConnection,
     Query(params): Query<Params>,
+    DatabaseConnection(db): DatabaseConnection,
```
Same here.

jplatte commented on frameworks/Rust/axum/src/main_pg.rs:
```diff
@@ -59,8 +59,8 @@ async fn fortunes(DatabaseConnection(conn): DatabaseConnection) -> impl IntoResp
 async fn updates(
-    DatabaseConnection(conn): DatabaseConnection,
     Query(params): Query<Params>,
+    DatabaseConnection(conn): DatabaseConnection,
```
Same here.

jplatte commented on frameworks/Rust/axum/Cargo.toml:
```diff
@@ -29,26 +29,27 @@ name = "axum-pg"
+tower = { version = "0.4.13", features = ["util"] }
+tower-http = { version = "0.3.5", features = ["set-header"] }
+yarte = "0.15.7"
+
```
unnecessary extra newline

```suggestion
```

jplatte reviewed (commented):
So it seems like you are misunderstanding dependency requirements in `Cargo.toml`. While they don't hurt, every version increase outside of the one for `axum` doesn't really make a difference, because a requirement like `futures = "0.3.21"` is equivalent to `futures = ">=0.3.21 <0.4"` ([docs](https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html)). Please run `cargo update` to actually get all of the latest crate versions (including of indirect dependencies).

romanstingler commented:
@jplatte 
I addressed your comments and did some cleanup based on clippy's suggestions.
I tested it locally, and it verified and run the benchmarks just fine.

offtopic: wo kann ich dir das Lehrgeld überweisen? :)

nbrady-techempower commented:
Is this good to be merged?

jplatte reviewed (approved):
> Is this good to be merged?

Yes.
